From 7629fe035d888dad196b3b10820f8d3073113d64 Mon Sep 17 00:00:00 2001
From: Hou Zhiqiang <B48286@freescale.com>
Date: Fri, 29 May 2015 16:29:35 +0800
Subject: [PATCH 168/451] arm64/ls1043a: modified the DT upon platform clock

1. Make all clocks to 400MHz referred from platform clock.
2. Add the 'ranges' and '#size-cells' properity for clockgen node.
3. fixed the '#clock_cell' properity for cluster1_clk node.

Signed-off-by: Hou Zhiqiang <B48286@freescale.com>
Signed-off-by: Mingkai Hu <Mingkai.Hu@freescale.com>
[Xulin: Original patch taken from
Linux-LS1043A-SDK-V0.4-SOURCE-20150826-yocto.iso]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi |   55 ++++++++++++------------
 1 files changed, 27 insertions(+), 28 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
index 41c6a5f..7e388fb 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
@@ -99,7 +99,7 @@
 		reg-names = "QuadSPI", "QuadSPI-memory";
 		interrupts = <0 99 0x4>;
 		clock-names = "qspi_en", "qspi";
-		clocks = <&platform_clk 1>, <&platform_clk 1>;
+		clocks = <400000000>, <400000000>;
 		big-endian;
 		amba-base = <0x40000000>;
 	};
@@ -494,10 +494,9 @@
 	};
 
 	clockgen: clocking@1ee1000 {
-		compatible = "fsl,qoriq-clockgen-2.0";
-		reg = <0x0 0x1ee1000 0x0 0x1000>;
+		ranges = <0x0 0x0 0x1ee1000 0x1000>;
 		#address-cells = <1>;
-		#size-cells = <0>;
+		#size-cells = <1>;
 		sysclk: sysclk {
 			compatible = "fixed-clock";
 			#clock-cells = <0>;
@@ -508,7 +507,7 @@
 		cga_pll1: pll1@800 {
 			compatible = "fsl,qoriq-core-pll-2.0";
 			#clock-cells = <1>;
-			reg = <0x800>;
+			reg = <0x800 0x10>;
 			clocks = <&sysclk>;
 			clock-output-names = "cga-pll1", "cga-pll1-div2",
 					"cga-pll1-div3", "cga-pll1-div4";
@@ -517,7 +516,7 @@
 		cga_pll2: pll2@820 {
 			compatible = "fsl,qoriq-core-pll-2.0";
 			#clock-cells = <1>;
-			reg = <0x820>;
+			reg = <0x820 0x10>;
 			clocks = <&sysclk>;
 			clock-output-names = "cga-pll2", "cga-pll2-div2",
 					"cga-pll2-div3", "cga-pll2-div4";
@@ -526,15 +525,15 @@
 		platform_clk: pll@c00 {
 			compatible = "fsl,qoriq-core-pll-2.0";
 			#clock-cells = <1>;
-			reg = <0xc00>;
+			reg = <0xc00 0x10>;
 			clocks = <&sysclk>;
 			clock-output-names = "platform-clk", "platform-clk-div2";
 		};
 
 		cluster1_clk: clk0c0@0 {
 			compatible = "fsl,qoriq-core-mux-2.0";
-			#clock-cells = <1>;
-			reg = <0x0>;
+			#clock-cells = <0>;
+			reg = <0x0 0x10>;
 			clock-names = "pll1cga", "pll1cga-div2", "pll1cga-div4",
 				      "pll2cga", "pll2cga-div2", "pll2cga-div4";
 			clocks = <&cga_pll1 0>, <&cga_pll1 1>, <&cga_pll1 2>,
@@ -556,7 +555,7 @@
 		reg = <0x0 0x2100000 0x0 0x10000>;
 		interrupts = <0 64 0x4>;
 		clock-names = "dspi";
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		spi-num-chipselects = <5>;
 		big-endian;
 		tcfq-mode;
@@ -570,7 +569,7 @@
 		reg = <0x0 0x2110000 0x0 0x10000>;
 		interrupts = <0 65 0x4>;
 		clock-names = "dspi";
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		spi-num-chipselects = <5>;
 		big-endian;
 		status = "disabled";
@@ -583,7 +582,7 @@
 		reg = <0x0 0x2180000 0x0 0x10000>;
 		interrupts = <0 56 0x4>;
 		clock-names = "i2c";
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 	};
 
 	i2c1: i2c@2190000 {
@@ -593,7 +592,7 @@
 		reg = <0x0 0x2190000 0x0 0x10000>;
 		interrupts = <0 57 0x4>;
 		clock-names = "i2c";
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		status = "disabled";
 	};
 
@@ -604,7 +603,7 @@
 		reg = <0x0 0x21a0000 0x0 0x10000>;
 		interrupts = <0 58 0x4>;
 		clock-names = "i2c";
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		status = "disabled";
 	};
 
@@ -615,7 +614,7 @@
 		reg = <0x0 0x21b0000 0x0 0x10000>;
 		interrupts = <0 59 0x4>;
 		clock-names = "i2c";
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		status = "disabled";
 	};
 
@@ -624,7 +623,7 @@
 		compatible = "fsl,ns16550", "ns16550";
 		reg = <0x00 0x21c0500 0x0 0x100>;
 		interrupts = <0 54 0x4>;
-		clock-frequency = <&platform_clk 1>;
+		clock-frequency = <400000000>;
 	};
 
 	duart1: serial@21c0600 {
@@ -632,7 +631,7 @@
 		compatible = "fsl,ns16550", "ns16550";
 		reg = <0x00 0x21c0600 0x0 0x100>;
 		interrupts = <0 54 0x4>;
-		clock-frequency = <&platform_clk 1>;
+		clock-frequency = <400000000>;
 	};
 
 	duart2: serial@21d0500 {
@@ -640,7 +639,7 @@
 		compatible = "fsl,ns16550", "ns16550";
 		reg = <0x0 0x21d0500 0x0 0x100>;
 		interrupts = <0 55 0x4>;
-		clock-frequency = <&platform_clk 1>;
+		clock-frequency = <400000000>;
 	};
 
 	duart3: serial@21d0600 {
@@ -648,7 +647,7 @@
 		compatible = "fsl,ns16550", "ns16550";
 		reg = <0x0 0x21d0600 0x0 0x100>;
 		interrupts = <0 55 0x4>;
-		clock-frequency = <&platform_clk 1>;
+		clock-frequency = <400000000>;
 	};
 
 	gpio1: gpio@2300000 {
@@ -705,7 +704,7 @@
 		compatible = "fsl,ls1021a-lpuart";
 		reg = <0x0 0x2960000 0x0 0x1000>;
 		interrupts = <0 49 0x4>;
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		clock-names = "ipg";
 		status = "disabled";
 	};
@@ -714,7 +713,7 @@
 		compatible = "fsl,ls1021a-lpuart";
 		reg = <0x0 0x2970000 0x0 0x1000>;
 		interrupts = <0 50 0x4>;
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		clock-names = "ipg";
 		status = "disabled";
 	};
@@ -723,7 +722,7 @@
 		compatible = "fsl,ls1021a-lpuart";
 		reg = <0x0 0x2980000 0x0 0x1000>;
 		interrupts = <0 51 0x4>;
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		clock-names = "ipg";
 		status = "disabled";
 	};
@@ -732,7 +731,7 @@
 		compatible = "fsl,ls1021a-lpuart";
 		reg = <0x0 0x2990000 0x0 0x1000>;
 		interrupts = <0 52 0x4>;
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		clock-names = "ipg";
 		status = "disabled";
 	};
@@ -741,7 +740,7 @@
 		compatible = "fsl,ls1021a-lpuart";
 		reg = <0x0 0x29a0000 0x0 0x1000>;
 		interrupts = <0 53 0x4>;
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		clock-names = "ipg";
 		status = "disabled";
 	};
@@ -757,7 +756,7 @@
 		compatible = "fsl,imx21-wdt";
 		reg = <0x0 0x2ad0000 0x0 0x10000>;
 		interrupts = <0 83 0x4>;
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 		clock-names = "wdog";
 		big-endian;
 	};
@@ -774,8 +773,8 @@
 		dma-channels = <32>;
 		big-endian;
 		clock-names = "dmamux0", "dmamux1";
-		clocks = <&platform_clk 1>,
-		         <&platform_clk 1>;
+		clocks = <400000000>,
+			 <400000000>;
 	};
 
 	usb0: usb3@2f00000 {
@@ -802,7 +801,7 @@
 		compatible = "fsl,ls1021a-ahci";
 		reg = <0x0 0x3200000 0x0 0x10000>;
 		interrupts = <0 69 0x4>;
-		clocks = <&platform_clk 1>;
+		clocks = <400000000>;
 	};
 
 	qdma: qdma@8380000 {
-- 
1.7.5.4

